radicle-tracker/src/lib.rs

```diff
@@ -0,0 +1,63 @@
+use std::hash;
+
+pub mod thread;
+use thread::*;
+
+pub struct Issue<Author> {
+    pub author: Author,
+    pub title: String,
+    pub thread: Thread<Comment<Author>>,
+    pub meta: (), // TODO(fintan): fill in meta data
+}
+
+impl<Author> Issue<Author> {
+    /// ```
+    /// use radicle_tracker::Issue;
+    ///
+    /// let issue = Issue::new(
+    ///     String::from("Monadic"),
+    ///     String::from("Buggy Boeuf"),
+    ///     String::from("We have bugs in our boeuf"),
+    /// );
+    ///
+    /// assert_eq!(issue.author, String::from("Monadic"));
+    /// assert_eq!(issue.title, String::from("Buggy Boeuf"));
+    /// assert_eq!(issue.thread.first().content, String::from("We have bugs in our boeuf"));
+    /// assert!(issue.thread.first().author == issue.author);
+    /// assert_eq!(issue.meta, ());
+    /// ```
+    pub fn new(author: Author, title: String, content: String) -> Self
+    where
+        Author: Clone + Eq,
+    {
+        let comment = Comment {
+            author: author.clone(),
+            content,
+        };
+
+        Issue {
+            author,
+            title,
+            thread: Thread::new(comment).0,
+            meta: (),
+        }
+    }
+
+    pub fn reply_to_thread(&mut self, path: Path, comment: Comment<Author>) -> Result<Path, ()> {
```

Comment:
Some doctests for the reply would help illustrate how that works. Esp. keen on understanding the `Path` part. I imagine that to be quintessential to refer into threads in issues.

Reply:
For sure, that's the aim for the next iteration :grin: